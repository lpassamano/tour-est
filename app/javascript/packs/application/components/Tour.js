import React, { Component } from "react";
import { connect } from "react-redux";
import * as tourActions from "../redux/tours/actions";
import * as tourSelectors from "../redux/tours/selectors";

class Tour extends Component {
  componentDidMount() {
    this.props.getTour(this.props.tourId);
  }

  render() {
    const { isFetching, tour } = this.props;
    console.log("tour", tour);
    if (isFetching || !tour) {
      return <p>loading... please wait!</p>;
    }

    return (
      <div>
        <h2>{tour.title}</h2>
        <p>{tour.estimated_time}</p>
        <p>{tour.description}</p>
        <p>{tour.starting_point}</p>
        <p>{tour.directions}</p>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  tour: tourSelectors.getTour(state, ownProps.tourId),
  isFetching: tourSelectors.isFetching(state)
});

const mapDispatchToProps = { getTour: tourActions.getTour };

const enhance = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default enhance(Tour);
