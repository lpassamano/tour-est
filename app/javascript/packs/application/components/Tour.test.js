import { Tour } from "./Tour";
import React from "react";
import { shallow } from "enzyme";

describe("<Tour />", () => {
  test("render", () => {
    const tour = {
      title: "Philly Tour",
      estimated_time: "2 hours",
      description: "Walking tour of Philly's architecture and public art",
      starting_point: "City Hall",
      directions: "Meet at the West entrance"
    };
    const getTour = jest.fn();
    const component = shallow(<Tour tour={tour} getTour={getTour} />);
    expect(component).toMatchSnapshot();
  });
});
