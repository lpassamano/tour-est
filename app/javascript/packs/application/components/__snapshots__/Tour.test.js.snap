// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tour /> render 1`] = `
<div>
  <h2>
    Philly Tour
  </h2>
  <p>
    2 hours
  </p>
  <p>
    Walking tour of Philly's architecture and public art
  </p>
  <p>
    City Hall
  </p>
  <p>
    Meet at the West entrance
  </p>
</div>
`;
